File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key = lambda x: x[1])
        res = 1
        j = 1
        i = 0
        while j < len(points):
            if points[j][0] <= points[i][1]:
                j += 1
                continue
            res += 1
            i = j
        return res
            

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, Comparator.comparingInt(a -> a[1]));
        int res = 1;
        int i = 0;
        int j = 1;
        while(j < points.length){
            if(points[j][0] <= points[i][1]){
                j++;
                continue;
            }
            
            res++;
            i = j;  
        }
        return res;
    }
}